Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -916,4 +910,15 @@ std::partial_ordering AXTextMarker::partialOrderByTraversal(const AXTextMarker&
 }
 #endif // ENABLE(AX_THREAD_TEXT_APIS)
 
+TextMarkerData RawTextMarkerData::toTextMarkerData() const
+{
+    ObjectIdentifier<AXIDType> axTreeID(treeID);
+    WeakPtr cache = AXTreeStore<AXObjectCache>::axObjectCacheForID(axTreeID);
+    // Make sure the node is not stale before storing it in a WeakPtr.
+    Node* validNode = nullptr;
+    if (node && cache && cache->isNodeInUse(*node))
```

Comment:
To address the failures, I had to add this `isNodeInUse()` check to make sure the `node` pointer is not stable after deserializing, before storing in a WeakPtr. `AXObjectCache::isInUse()` is how accessibility code currently does stale node detection.